### PR TITLE
E1.33 cherry pick the second

### PR DESCRIPTION
**Pull request by peternewman**
The next sub-part of #1841

### CODE REVIEW

File: libs/acn/RootPDU.h [hunk 1]
```diff
@@ -33,8 +33,8 @@ namespace acn {
 
 class RootPDU: public PDU {
  public:
-  explicit RootPDU(unsigned int vector):
-    PDU(vector),
+  explicit RootPDU(unsigned int vector, bool force_length_flag = false):
+    PDU(vector, FOUR_BYTES, force_length_flag),
```
Comment (kripton) [inline libs/acn/RootPDU.h:37]: Just a question: What are the "FOUR_BYTES"? Some kind of magic/FourCC?

File: tools/e133/E133StatusHelper.cpp [hunk 2]
```diff
@@ -96,5 +96,51 @@ string StatusCodeToString(E133StatusCode status_code) {
   }
   return "Unknown E1.33 Status Code";
 }
+
+
+bool IntToConnectStatusCode(uint16_t input,
+                            E133ConnectStatusCode *connect_status_code) {
+  switch (input) {
+    case ola::e133::CONNECT_OK:
```
Comment (kripton) [inline tools/e133/E133StatusHelper.cpp:104]: how about:
```
 case ola::e133::CONNECT_OK:
 case ola::e133::CONNECT_SCOPE_MISMATCH:
.......
      *connect_status_code = input;
      return true;
```
?

**Review by kripton — APPROVED** [review]
See inline comments. Some might be stale due to changes pushed during my review :)

File: include/ola/acn/ACNPort.h [hunk 3]
```diff
@@ -61,6 +61,11 @@ const uint16_t ACN_PORT = 5568;
  */
 const uint16_t E133_PORT = 5569;
 
+/**
+ * @brief The port used for E1.33 LLRP communication.
+ */
+const uint16_t LLRP_PORT = 5569;
```
[the remark this answers is not in the document]
Comment (peternewman) [reply include/ola/acn/ACNPort.h:67]: I think E133_PORT becomes redundant looking at where it's used, and they essentially reused the port they'd chosen/had allocated.

I'm inclined to leave them separate for now and I've added a task to the primary PR to remove all the redundant code from the pre-standards version.

File: libs/acn/RootPDU.h (hunk 1 above)
Comment (peternewman) [reply libs/acn/RootPDU.h:37]: Not really, it's just an internal enum for the width of the vector, defined here:
https://github.com/OpenLightingProject/ola/blob/e7b403927ad462d6ba4eed7d72cdd3e1fb1fa4e8/libs/acn/PDU.h#L39-L43

Just a bit more user-friendly than magic numbers like 4 appearing all over the code with no context.

File: tools/e133/E133StatusHelper.cpp (hunk 2 above)
Comment (peternewman) [reply tools/e133/E133StatusHelper.cpp:104]: That's a very elegant solution.

Unfortunately that code doesn't solely exist in E1.33 land. Do you fancy a clean-up of this (once merged) and the other similar code (e.g. there's some in https://github.com/OpenLightingProject/ola/blob/master/common/rdm/RDMHelper.cpp )?

We'd also need to add OLA_FALLTHROUGH to cover us from other compiler warnings, but looks neat.

**Comment by peternewman** [comment]
> See inline comments.

Thanks.

> Some might be stale due to changes pushed during my review :)

Yeah sorry it was a bit lumpy! Do you want to check my replies and double-check nothing else snuck through whilst you **were** looking?

File: libs/acn/RootPDU.h (hunk 1 above)
Comment (kripton) [reply libs/acn/RootPDU.h:37]: Okay, thanks for explaining :)

File: tools/e133/E133StatusHelper.cpp (hunk 2 above)
Comment (kripton) [reply tools/e133/E133StatusHelper.cpp:104]: I can have a look, yes. Ping me once you merged it!

**Comment by kripton** [comment]
> > Some might be stale due to changes pushed during my review :)
> 
> Yeah sorry it was a bit lumpy! Do you want to check my replies and double-check nothing else snuck through whilst you **were** looking?

No problem. It should be fine, github makes it quite easy to track what changed after I started reviewing so go ahead merging it :+1:

File: tools/e133/E133StatusHelper.cpp (hunk 2 above)
Comment (peternewman) [reply tools/e133/E133StatusHelper.cpp:104]: It's in now @kripton but I assume you got an automatic notification anyway...

**Pull request merged**